driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -12,10 +16,52 @@ use futures::{
 use orderbook::Orderbook;
 use std::sync::{Arc, Mutex};
 
-// TODO:
-// `pub struct UpdatingOrderbook` that also owns an `Arc<Mutex<Orderbook>>` and creates a
-// background thread running `UpdatingOrderbookThread::update_with_events`.
-// This way `UpdatingOrderbookThread` is independently testable.
+/// An event based orderbook that automatically updates itself with new events from the contract.
+#[derive(Debug)]
+pub struct UpdatingOrderbook {
+    orderbook: Arc<Mutex<Orderbook>>,
+    // When this struct is dropped this sender will be dropped which makes the updater thread stop.
+    _channel: oneshot::Sender<()>,
+}
+
+impl UpdatingOrderbook {
+    pub fn new(
+        _contract: &impl StableXContract,
+        block_timestamp: impl BlockTimestamp + Send + 'static,
+    ) -> Self {
+        let orderbook = Arc::new(Mutex::new(Orderbook::default()));
+        let orderbook_clone = orderbook.clone();
+        let (sender, receiver) = oneshot::channel();
```

Comment:
naming nit: maybe call the sender and receiver based on what they are used for? Maybe `(exit_tx, exit_rx)`?